src/Data/CAS/RocksDB.hs

```diff
@@ -314,15 +317,39 @@ newTable
     -> Codec v
     -> Codec k
     -> [B.ByteString]
+    -> IO (RocksDbTable k v)
+newTable db valCodec keyCodec namespace = do
+    table <- evaluate (unregisteredNewTable db valCodec keyCodec namespace)
+    tableInsert (tablesTable db) namespace ()
+    return table
+{-# INLINE newTable #-}
```

Comment:
This change may cause issues. Currently, tables are just abstractions of key ranges and value encodings. They aren't database objects on their on. Consequently, creation a table is a pure and idempotent operation. Making it an non-idempotent IO operation would probably cause issues.

Reply:
Also, changing the database schema may cause compatibility issues that would have to be taken into consideration.

Reply:
> This change may cause issues. Currently, tables are just abstractions of key ranges and value encodings. They aren't database objects on their on. Consequently, creation a table is a pure and idempotent operation. Making it an non-idempotent IO operation would probably cause issues.

It's still idempotent from what I can tell. I noticed that the uses of `newTable` and `newCas` in chainweb-node happened to coincide with other resources being allocated, so putting it in `IO` isn't impossible.